src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -104,6 +97,25 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     blockTxsExecResult
   }
 
+  @tailrec
+  private[ledger] final def executePreparedTransactions(
+    signedTransactions: Seq[SignedTransaction], world: InMemoryWorldStateProxy,
+    blockHeader: BlockHeader, signedTransactionValidator: SignedTransactionValidator,
+    acumGas: BigInt = 0, acumReceipts: Seq[Receipt] = Nil, executed: Seq[SignedTransaction] = Nil)
+  : Either[TxsExecutionError, (BlockResult, Seq[SignedTransaction])] = {
+
+    val result = executeTransactions(signedTransactions, world, blockHeader, signedTransactionValidator, acumGas, acumReceipts)
+
+    result match {
+      case Left(TxsExecutionError(stx, StateBeforeFailure(worldState, gas, receipts), reason)) if signedTransactions.length > 1 =>
```

Comment:
What about if `signedTransactions.length == 1` and fails? I think it won't be able to create an empty block, do we want that behavior ?